src/main/kotlin/ecommerce/service/CartService.kt

```diff
@@ -0,0 +1,73 @@
+package ecommerce.service
+
+import ecommerce.dto.cart.CartDTO
+import ecommerce.dto.cartProduct.CartProductResponseDTO
+import ecommerce.enums.CartAction
+import ecommerce.exception.EntityNotFoundException
+import ecommerce.repository.CartProductRepository
+import ecommerce.repository.CartRepository
+import ecommerce.repository.CartStatisticsRepository
+import ecommerce.repository.ProductRepository
+import org.springframework.stereotype.Service
+
+@Service
+class CartService(
+    private val cartRepository: CartRepository,
+    private val cartProductRepository: CartProductRepository,
+    private val productRepository: ProductRepository,
+    private val cartStatisticsRepository: CartStatisticsRepository,
+) {
+    fun getCartProducts(userID: Long?): List<CartProductResponseDTO> {
+        val cart = getUserCart(userID)
+        return cartProductRepository.getCartProducts(cart.id)
+    }
+
+    fun addProductToCart(
+        userID: Long?,
+        productID: Long,
+    ): Long {
+        var id: Long
```

Comment:
To reduce side effects, please keep mutable variables as close as possible to where they are used. Also, consider ways to avoid using mutable variables altogether 🙂